lib/components/MiningDashboard.js

```diff
@@ -70,19 +66,20 @@ class MiningDashboard extends React.PureComponent {
 
   async mineBlocks(e) {
     e.preventDefault();
-    const { savedCoinbase, numBlocks } = this.state;
+    const { numBlocks } = this.state;
+    const { coinbaseAddress } = this.props;
     try {
-      if (!savedCoinbase) {
+      if (!coinbaseAddress) {
         alert('Must set a coinbase address to mine');
       } else {
         this.setState({ mining: true });
```

Comment:
Any particular reason to use `then` over `await`?

Reply:
Good question. I can't remember exactly what the thought process was to be honest. I think it either has to do with behavior around `setState` or the fact that the rpc `execute` calls only return a confirmation from the server that the call was received, not an actual response once the operation was done. I might have been trying to play with timeouts or something. I'll try and update to use async/await since I think that's more readable.

Reply:
Either one of those or it could have been to avoid nested try/catches, either way, doesn't seem worth it.

Reply:
Updated, didn't seem to cause any issues. Thanks for catching!